scripts/utils.js

```diff
@@ -240,46 +241,77 @@ function buildAuthorHeader(mainEl) {
   div.prepend(authorHeader);
 }
 
-async function buildArticleHeader(el) {
+/**
+ * * @param {HTMLElement} element
+ * * @param {string} targetTag, like 'ul' or 'div'
+ * result: return the new element with inner content of the element, desired tag and css class
+ */
+export function changeHTMLTag(el, targetTag, properties = {}) {
+  const newEl = document.createElement(targetTag);
+  const innerContent = el.innerHTML;
+
+  Object.keys(properties).forEach((key) => {
+    newEl.setAttribute(key, properties[key]);
+  });
+
+  newEl.innerHTML = innerContent;
+  el.replaceWith(newEl);
+
+  return newEl;
+}
+
+async function buildArticleHeroBanner(el) {
   const miloLibs = getLibs();
-  const { getMetadata, getConfig } = await import(`${miloLibs}/utils/utils.js`);
+  const { getMetadata } = await import(`${miloLibs}/utils/utils.js`);
   const { loadTaxonomy, getLinkForTopic, getTaxonomyModule } = await import(`${miloLibs}/blocks/article-feed/article-helpers.js`);
   if (!getTaxonomyModule()) {
     await loadTaxonomy();
   }
+
   const div = document.createElement('div');
-  // div.setAttribute('class', 'section');
+
+  const tag = getMetadata('article:tag');
+  const category = tag || 'News';
+  const categoryTag = getLinkForTopic(category); // category link
+
   const h1 = el.querySelector('h1');
+  const description = getMetadata('description');
   const picture = el.querySelector('a[href*=".mp4"], picture');
   const caption = getImageCaption(picture);
   const figure = document.createElement('div');
   figure.append(picture, caption);
-  const tag = getMetadata('article:tag');
-  const category = tag || 'News';
-  const author = getMetadata('author') || 'Adobe Communications Team';
-  const { codeRoot } = getConfig();
-  const authorURL = getMetadata('author-url') || (author ? `${codeRoot}/authors/${author.replace(/[^0-9a-z]/gi, '-').toLowerCase()}` : null);
-  const publicationDate = getMetadata('publication-date');
 
-  const categoryTag = getLinkForTopic(category);
+  const tagEl = document.createElement('p');
+  tagEl.textContent = category;
 
-  const articleHeaderBlockEl = buildBlock('article-header', [
-    [`<p>${categoryTag}</p>`],
-    [h1],
-    [`<p>${authorURL ? `<a href="${authorURL}">${author}</a>` : author}</p>
-      <p>${publicationDate}</p>`],
-    [figure],
+  const marqueeEl = buildBlock('marquee', [
+    ['<p>transparent</p>'],
+    [
+      {
+        elems: [
+          categoryTag,
+          h1,
+          `<p>${description}</p>`,
+        ],
+      },
+      picture,
+    ],
   ]);
-  div.append(articleHeaderBlockEl);
+
+  marqueeEl.classList.add('split', 'medium', 'light', 'circle-gradient-pink', 'article-hero-banner');
```

Comment:
This is a great default. But, if I'mm understanding this, ALL article pages will be a split marquee with that circle gradient from now on, is that correct?

Does Kristine understand this? I ask because having an opt out to the previous version where they can have a full width image or video would probably be a good thing. There's always questions about what goes into the article header and if they can adjust it.

Reply:
Thanks for the flag! We'll chat with Kristine and see how we'll need to update the code for the customization support.

Reply:
Thanks for the great suggestion. I'm thinking about 2 metadata properties for the customization. 

One is the `layout`, we can make the sidebar layout as default one and add in option for rolling back to full width / single column article.

Another one is `hero gradient` or maybe `theme gradient`. We can see which approach in the screenshot do we prefer (option 1 vs option 2)

<img width="1172" alt="Screen Shot 2024-09-25 at 5 08 48 PM" src="https://github.com/user-attachments/assets/4f8b11d5-4c42-4d7a-8312-d7b14126b1dc">

But we can chat more about this during the meeting next week for the customization support

Reply:
one thought as it relates to `metadata` our `section-metada` already has a key of `layout`, which is a little too vague in hindsight, but regardless it acts as grid bootstrap for when using with `style: two-up` you can add `layout: 1 | 2` for a fractional division. Just noting that so we don't confuse that keys intent. This might be the place where you could use a template key

Reply:
Hey @rgclayton, sorry to bother you. Not sure if it's the best way to contact you as we don't have a slack channel setup yet (I think here it's at least better than email)

I've updated the option to only apply the new layout on another branch `redesign/article`. Would you mind taking a look and see if this approach in the Sharepoint document side looks good to you? Or if there're other updates needed? 

Keeping its styles by default (only added progress bar):  
- https://redesign-article--blog--adobecom.hlx.page/en/publish/2024/09/19/sustainability-constructive-perspectives

New layout (only show up when `Page Template` is set as `sidebar` in page metadata):
- https://redesign-article--blog--adobecom.hlx.page/en/drafts/snowie-testing/blog-detail/article-with-banner
- The user can pass in 2 hexcodes for custom colors while maintaining the circle-gradient pattern background, like `circle-gradient(#F8CCFB, #FBBAD8)`

If it looks good, I think the next step would be find out the articles with possible variable content & apply the identifier to test if it works well with all possible article content. 

Thank you so much

Reply:
@snowiewdev,

I think i like this. Falls inline with how most articles get authored on the blog.

> The user can pass in 2 hexcodes for custom colors while maintaining the circle-gradient pattern background, like circle-gradient(#F8CCFB, #FBBAD8). If nothing is passed in at the Hero Background field, it'll use the default circle gradient.

Related to this. Can the author add an image that shows up as a background? I can see that being an ask and viable option. Along with that, all blocks have a dark/light theme, applied as a variant, that changes the text color to appropriately sit on top of a dark or light background. Does that work with this new article header?

Reply:
It also looks like the CLS issues have been resolved? I'm not seeing them any more when running the page through pagespeed.web.dev.

Reply:
@rgclayton Thanks for the valuable insight & feedback! Updated staging site here:
https://redesign-article--blog--adobecom.hlx.page/en/drafts/snowie-testing/blog-detail/article-with-banner

For now users can add `article-hero-marquee` block to the page, they can pass in marquee variant like `Article Hero Marquee (light)` or `Article Hero Marquee (dark)` for the dark/light theme setting. They can custom `background` via passing in `image`, `circle-gradient(#F8CCFB, #FBBAD8)`, `color` in the block directly.

The block will automatically pull in content based on metadata & image in the article so editor don't need to enter the information again. Let me know if that looks good or more updates are needed. Thank you